mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -318,7 +331,9 @@ def _download_file(self, remote_file_path, local_path):
             self.run_relative_artifact_repo_root_path, remote_file_path
         )
         read_credentials = self._get_read_credentials(self.run_id, run_relative_remote_file_path)
-        self._download_from_cloud(read_credentials.credentials, local_path)
+        return self.thread_pool.submit(
+            self._download_from_cloud, read_credentials.credentials, local_path,
+        )
```

Comment:
@apurva-koti I don't think this will work because the implementation of `ArtifactRepository.download_artifacts()` assumes that this method is synchronous (https://github.com/mlflow/mlflow/blob/df8ba4e440a260c4b78294a735be0bfe1fe94902/mlflow/store/artifact/artifact_repo.py#L90).

As a result, calls such as `mlflow.pyfunc.load_model()` that attempt to load file contents into memory immediately after the invocation of `download_artifacts()` will likely fail, since the artifacts aren't actually present.

Additionally, this approach only provides parallelism on a per-directory level.

Instead, we should override the entire `download_artifacts()` method in `DatabricksArtifactRepository` and provide a custom implementation.

Reply:
Ohh yeah that makes sense. Let's override `download_artifacts` to keep all this special parallelism contained within `databricks_artifact_repo`